SpatialGDK/ExternalSchemaCodegen/Programs/Improbable.CodeGen.Unreal/Serialization.cs

```diff
@@ -0,0 +1,443 @@
+                    return $"Schema_GetUint32({schemaObjectName}, {fieldId})";
+                case PrimitiveType.Uint64:
+                    return $"Schema_GetUint64({schemaObjectName}, {fieldId})";
+                case PrimitiveType.Sint32:
+                    return $"Schema_GetSint32({schemaObjectName}, {fieldId})";
+                case PrimitiveType.Sint64:
+                    return $"Schema_GetSint64({schemaObjectName}, {fieldId})";
+                case PrimitiveType.Fixed32:
+                    return $"Schema_GetFixed32({schemaObjectName}, {fieldId})";
+                case PrimitiveType.Fixed64:
+                    return $"Schema_GetFixed32({schemaObjectName}, {fieldId})";
+                case PrimitiveType.Sfixed32:
+                    return $"Schema_GetSfixed32({schemaObjectName}, {fieldId})";
+                case PrimitiveType.Sfixed64:
+                    return $"Schema_GetSfixed64({schemaObjectName}, {fieldId})";
+                case PrimitiveType.Bool:
+                    return $"Schema_GetBool({schemaObjectName}, {fieldId})";
+                case PrimitiveType.Float:
+                    return $"Schema_GetFloat({schemaObjectName}, {fieldId})";
+                case PrimitiveType.Double:
+                    return $"Schema_GetDouble({schemaObjectName}, {fieldId})";
+                case PrimitiveType.EntityId:
+                    return $"Schema_GetEntityId({schemaObjectName}, {fieldId})";
+                case PrimitiveType.Bytes:
+                    return $"::improbable::utils::GetBytes({schemaObjectName}, {fieldId})";
+                case PrimitiveType.String:
+                    return $"::improbable::utils::GetString({schemaObjectName}, {fieldId})";
+                case PrimitiveType.Invalid:
+                default:
+                    throw new InvalidOperationException("Trying to serialize invalid PrimitiveType");
+            }
+        }
+
+        private static string GetValueTypeCount(ValueTypeReference type, string schemaObjectName, string fieldId)
+        {
+            switch (type.ValueTypeSelector)
+            {
+                case ValueType.Primitive:
+                    return GetPrimitiveCount(type.Primitive, schemaObjectName, fieldId);
+                case ValueType.Type:
+                    return $"Schema_GetObjectCount({ schemaObjectName}, { fieldId})";
+                case ValueType.Enum:
+                    return $"Schema_GetEnumCount({ schemaObjectName}, { fieldId})";
+                default:
+                    throw new InvalidOperationException("Trying to deserialize invalid ValueTypeReference");
+            }
+        }
+
+        private static string GetPrimitiveCount(PrimitiveType primitive, string schemaObjectName, string fieldId)
+        {
+            switch (primitive)
+            {
+                case PrimitiveType.Int32:
+                    return $"Schema_GetInt32Count({ schemaObjectName}, { fieldId})";
```

Comment:
Nit: extra space before `fieldId`
Collapse all the numeric types down into one return statement